### PR TITLE
Split repository API for module and DS management. Refactor utility usage

### PR DESCRIPTION
**Pull request by kaizimmerm**
One more step in repository refactoring (see #197). I split software and distribution set management for type and instance management (more splits will follow e.g. TagManagement).

Another topic that is constantly bugging many hawkBit developers is the chaos we introduced with using a wide array of utility libraries. You can basically see in code which dev prefers apache or spring utils or Apache commons or JDK stuff.

This cannot be fixed over night. But I documented in the contributors guide a recommended priority and refactored the code base by means of the "low hanging fruits" to implement the new guideline.

I think in the long run we should be able to get rid of Apache commons and reduce guava to use cases  where it actually makes sense. Spring utils imho can be used freely, hawkBit is based on Spring and will continue to do so.

Let me know what you think.

Reviewers: @SirWayne @melanieretter

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +21,27 @@ Please read this if you intend to contribute to the project.
 * Sonarqube:
   * Our rule set can be found [here](https://sonar.ops.bosch-iot-rollouts.com/projects) with navigating to the tab "Quality Profiles", selecting "hawkBit", and then selecting "Actions" - "Back up"
 
+### Utility library usage
+
+hawkBit has currently both [guava](https://github.com/google/guava) and [Apache commons lang](https://commons.apache.org/proper/commons-lang/) on the classpath in several of its modules. However, we see introducing to many utility libraries problematic as we force these as transitive dependencies on hawkBit users. We in fact are looking into reducing them in future not adding new ones.
```
Comment (melanieretter) [inline CONTRIBUTING.md]: ... introducing **too** many...

**Comment by kaizimmerm** [comment]
[CQ13563](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=13563) for using caffeine cache instead of guava. Allows to keep guava out of core. Besides Boot will switch to Caffeine as default anyhow.

File: examples/hawkbit-example-mgmt-feign-client/src/main/java/org/eclipse/hawkbit/mgmt/client/resource/builder/DistributionSetBuilder.java [hunk 2]
```diff
@@ -114,7 +113,7 @@ public DistributionSetBuilder description(final String description) {
      * @return a list of {@link MgmtDistributionSetRequestBodyPost}
      */
     public List<MgmtDistributionSetRequestBodyPost> buildAsList(final int offset, final int count) {
-        final ArrayList<MgmtDistributionSetRequestBodyPost> bodyList = Lists.newArrayList();
```
Comment (kaizimmerm) [inline examples/hawkbit-example-mgmt-feign-client/src/main/java/org/eclipse/hawkbit/mgmt/client/resource/builder/DistributionSetBuilder.java]: Interface

File: hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java [hunk 3]
```diff
@@ -240,7 +239,9 @@ private void updateActionStatus(final Message message) {
         final Action action = checkActionExist(message, actionUpdateStatus);
 
         final List<String> messages = actionUpdateStatus.getMessage();
-        if (ArrayUtils.isNotEmpty(message.getMessageProperties().getCorrelationId())) {
+
+        if (message.getMessageProperties().getCorrelationId() != null
```
Comment (kaizimmerm) [inline hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java]: Extract into method

File: hawkbit-security-core/src/test/java/org/eclipse/hawkbit/util/IpUtilTest.java [hunk 4]
```diff
@@ -8,7 +8,6 @@
  */
 package org.eclipse.hawkbit.util;
 
-import static com.google.common.net.HttpHeaders.X_FORWARDED_FOR;
```
Comment (kaizimmerm) [inline hawkbit-security-core/src/test/java/org/eclipse/hawkbit/util/IpUtilTest.java]: keep guava but on text scope

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/SPUIComponentProvider.java [hunk 5]
```diff
@@ -199,7 +198,7 @@ public static Panel getDistributionSetInfo(final DistributionSet distributionSet
      * @return Label
      */
     public static Label createNameValueLabel(final String label, final String... values) {
-        final String valueStr = StringUtils.join(Arrays.asList(values), " ");
+        final String valueStr = StringUtils.arrayToCommaDelimitedString(values);
```
Comment (melanieretter) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/SPUIComponentProvider.java]: Now the String is comma separated. The original was separated by a space character.

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDownloadArtifactResource.java [hunk 6]
```diff
@@ -39,7 +39,7 @@
 public class MgmtDownloadArtifactResource implements MgmtDownloadArtifactRestApi {
 
     @Autowired
-    private SoftwareManagement softwareManagement;
+    private SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDownloadArtifactResource.java]: Rename 'softwareManagement' to 'softwareModuleManagement'

File: hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java [hunk 7]
```diff
@@ -82,7 +82,7 @@
     private ControllerManagement controllerManagement;
 
     @Autowired
-    private SoftwareManagement softwareManagement;
+    private SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java]: Rename `softwareManagement `to 'softwareModuleManagement'

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDistributionSetResource.java [hunk 8]
```diff
@@ -63,7 +63,7 @@
     private static final Logger LOG = LoggerFactory.getLogger(MgmtDistributionSetResource.class);
 
     @Autowired
-    private SoftwareManagement softwareManagement;
+    private SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDistributionSetResource.java]: Rename `softwareManagement `

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDownloadArtifactResource.java (hunk 6 above)
Comment (SirWayne) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDownloadArtifactResource.java]: Rename `softwareManagement `

File: hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java [hunk 9]
```diff
@@ -57,7 +57,7 @@
     private ArtifactManagement artifactManagement;
 
     @Autowired
-    private SoftwareManagement softwareManagement;
+    private SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtSoftwareModuleResource.java]: Rename `softwareManagement `

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetTypeBuilder.java [hunk 10]
```diff
@@ -21,9 +21,9 @@
  */
 public class JpaDistributionSetTypeBuilder implements DistributionSetTypeBuilder {
 
-    private final SoftwareManagement softwareManagement;
+    private final SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetTypeBuilder.java]: Rename `softwareManagement`

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetTypeCreate.java [hunk 11]
```diff
@@ -26,9 +26,9 @@
 public class JpaDistributionSetTypeCreate extends AbstractDistributionSetTypeUpdateCreate<DistributionSetTypeCreate>
         implements DistributionSetTypeCreate {
 
-    private final SoftwareManagement softwareManagement;
+    private final SoftwareModuleManagement softwareManagement;
```
Comment (SirWayne) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/builder/JpaDistributionSetTypeCreate.java]: Rename `softwareManagement `

**Pull request merged**